File: frontend/lib/core/models/meal_plan.dart
class MealPlan {
  final String id;
  final String name;
  final String foodType;
  final int mealsPerDay;
  final double monthlyRate;
  final double extraMealRate;
  final bool isActive;

  const MealPlan({
    required this.id,
    required this.name,
    required this.foodType,
    required this.mealsPerDay,
    required this.monthlyRate,
    this.extraMealRate = 0,
    required this.isActive,
  });

  factory MealPlan.fromJson(Map<String, dynamic> json) => MealPlan(
        id: json['id'] as String,
        name: json['name'] as String,
        foodType: json['food_type'] as String,
        mealsPerDay: json['meals_per_day'] as int,
        monthlyRate: double.parse(json['monthly_rate'].toString()),
        extraMealRate:
            double.parse((json['extra_meal_rate'] ?? '0').toString()),
        isActive: json['is_active'] as bool,
      );
}

File: backend/app/schemas/meal_plan.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.meal_plan import FoodType


class PlanCreate(BaseModel):
    name: str = Field(min_length=1, max_length=50)
    food_type: FoodType
    meals_per_day: int = Field(ge=2, le=3)
    monthly_rate: Decimal = Field(ge=0, decimal_places=2)
    extra_meal_rate: Decimal = Field(default=Decimal("0.00"), ge=0, decimal_places=2)


class PlanUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=50)
    monthly_rate: Decimal | None = Field(default=None, ge=0, decimal_places=2)
    extra_meal_rate: Decimal | None = Field(default=None, ge=0, decimal_places=2)
    is_active: bool | None = None


class PlanResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    food_type: FoodType
    meals_per_day: int
    monthly_rate: Decimal
    extra_meal_rate: Decimal
    is_active: bool
    created_at: datetime
